aspnetcore/security/anti-request-forgery.md

```diff
@@ -1,18 +1,311 @@
+# Preventing Cross-Site Request Forgery (XSRF/CSRF) Attacks in ASP.NET Core
+
+[Steve Smith](http://ardalis.com/), [Fiyaz Hasan](https://twitter.com/FiyazBinHasan), David Paquette
+
+## What attack does anti-forgery prevent?
+
+Cross-site request forgery (also known as XSRF or CSRF, pronounced *see-surf*) is an attack against web-hosted applications whereby a malicious web site can influence the interaction between a client browser and a web site that trusts that browser. These attacks are made possible because web browsers send authentication tokens automatically with every request to a web site. This form of exploit is also known as a *one-click attack* or as *session riding*, because the attack takes advantage of the user's previously authenticated session.
+
+An example of a CSRF attack:
+
+1. A user logs into `www.example.com`, using forms authentication.
+
+2. The server authenticates the user and issues a response that includes an authentication cookie.
+
+3. The attacker uses social engineering to trick the user into clicking on a link to a malacious site.
+
+   This malicious site contains the following HTML form:
+
+<!-- literal_block {"xml:space": "preserve", "language": "html", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+```html
+   <h1>You Are a Winner!</h1>
+     <form action="http://example.com/api/account" method="post">
+       <input type="hidden" name="Transaction" value="withdraw" />
+       <input type="hidden" name="Amount" value="1000000" />
+     <input type="submit" value="Click Me"/>
+   </form>
+```
+
+Notice that the form action posts to the vulnerable site, not to the malicious site. This is the “cross-site” part of CSRF.
+
+4. The user clicks the submit button. The browser automatically includes the authentication cookie for the requested domain (the vulnerable site in this case) with the request.
+
+5. The request runs on the server with the user’s authentication context, and can do anything that an authenticated user is allowed to do.
+
+Although this example requires the user to click the form button, the malicious page could just as easily run a script (without any user interaction) that sends an AJAX request. Moreover, using SSL does not prevent a CSRF attack, because the malicious site can send an `https://` request. Some attacks can target site endpoints that respond to `GET` requests, in which case even an image tag can be used to perform the action (this form of attack is common on forum sites that permit images but block JavaScript). If your application uses `GET` requests to significantly change the state of the application, you should switch to `POST` if possible (in addition to protecting against CSRF attacks).
```

Comment:
We can but I don't know that we want to get into a dogma debate about what does or doesn't constitute REST (or whether following REST is or should be required). Let me know.